src/components/element-editor/conditional-rendering/conditonal-rendering.tsx

```diff
@@ -44,20 +49,20 @@ export const ConditionalRendering = () => {
 
     useEffect(() => {
         const element = focusedElement.element;
-        setChildEntryValues(element.conditionalRendering);
-    }, [focusedElement.element]);
-
-    useEffect(() => {
-        if (childEntryValues?.length >= 1) {
-            const updatedChildEntries = childEntryValues?.filter((child) => {
-                return Object.keys(child).some((id) => {
-                    return elements?.hasOwnProperty(id);
-                });
-            });
 
+        if (element.conditionalRendering?.length >= 1) {
+            const updatedChildEntries = element.conditionalRendering?.filter(
+                (child) => {
+                    return elements?.hasOwnProperty(child.internalId);
+                }
+            );
+            setValue("conditionalRendering", updatedChildEntries);
             setChildEntryValues(updatedChildEntries);
+        } else {
+            setValue("conditionalRendering", []);
+            setChildEntryValues([]);
         }
```

Comment:
no need to check for length, the filter will return an empty array if conditionalRendering is empty anyway